integration/tests/cook/test_basic.py

```diff
@@ -2626,6 +2626,68 @@ def test_pool_specific_quota_check_on_submit(self):
                 job_uuid, resp = util.submit_job(self.cook_url, pool=pool_name, mem=mem_over_quota)
                 self.assertEqual(422, resp.status_code, msg=resp.content)
 
+    @unittest.skipUnless(util.pool_quota_test_pool() is not None, 'Test requires a test pool.')
+    def test_pool_quota(self):
+        settings = util.settings(self.cook_url)
+        quotas_map = settings.get("pools", {}).get("quotas", [])
+        pool_name = util.pool_quota_test_pool()
+        match_quota = [ii["quota"] for ii in quotas_map if
+                       re.match(ii["pool-regex"], pool_name)]
+        # If the pool doesn't have the right quota constraints, fail the test.
+        logging.info("Quota: " + repr(match_quota))
+        if (len(match_quota) == 0):
+            self.fail(f"Pool {pool_name} lacks quota assignment.")
+        quota = match_quota[0]
+        logging.info(f"Pool quota: {quota}")
+        if (quota["count"] >= 10):
+            self.fail(f"Job count quota too large for test")
+        quota_count = quota["count"]
+        job_count = quota_count + 2
+        job_mem = 16
+        job_cpus = .01
+
+        if quota["mem"] < job_mem * (job_count + 3):
+            self.fail("Quota memory too small for test")
+        if quota["cpus"] < job_cpus * (job_count + 3):
+            self.fail("Quota cpus to small for test")
+
+        # Now lookup the user quota and make sure it fits and fail if otherwise.
+        user = self.determine_user()
+        resp = util.get_limit(self.cook_url, 'quota', user)
+        user_quota = resp.json()['pools'][pool_name]
+        logging.info(f"User quota {user_quota}")
+        if user_quota["count"] < job_count + 3:
+            self.fail("User quota count too small for test")
+        if user_quota["mem"] < job_mem * (job_count + 3):
+            self.fail("User quota memory too small for test")
+        if user_quota["cpus"] < job_cpus * (job_count + 3):
```

Comment:
Why `job_count + 3`?

Reply:
These checks were put in because you wanted the test to be self-protective against a future hypothetical environments where there was a very constrained user quota.

Not knowing exactly what hypothetical environment you were thinking of, I put  in a fudge factor of to handle roundoff or other accounting irregularities, but I'd be happy to put in any other number you think is appropriate.

Reply:
Why wouldn't it just be `job_count`? 

Maybe define a couple of variables:

```
total_cpus_requested = job_count * job_cpus
total_mem_requested = job_count * mem
```

And then make sure that the `user_quota` values are >= those? 

The point is to make sure that the test is passing specifically because of the global quota and not because of the user's quota. If we don't have these checks, this test could pass without any of your code changes.

Reply:
Making these changes.